src/tarantool_schema.c

```diff
@@ -198,40 +198,92 @@ schema_space_free(struct mh_schema_space_t *schema) {
 	}
 }
 
-int parse_field_type(const char *sfield, size_t sfield_len) {
-	if (sfield_len == 3) {
-		if (tolower(sfield[0]) == 's' &&
-		    tolower(sfield[1]) == 't' &&
-		    tolower(sfield[2]) == 'r')
-			return FT_STR;
-		if (tolower(sfield[0]) == 'n' &&
-		    tolower(sfield[1]) == 'u' &&
-		    tolower(sfield[2]) == 'm')
-			return FT_NUM;
+static const char *field_type_strs[] = {
```

Comment:
Just curious: should not we introduce an array of known field types and an array of known index part types as separate arrays? AFAIR, the latter is the subset of the former. There is nothing critical if we'll accept, say, `any` as an index part type, but why not to accept anything so?

The idea is interesting. Since we don't use (correct me if I'm wrong) those types, we can skip validation of them. It also may help us with forward compatibility: when a new type will be added (say, `datetime`), the connector will continue work with spaces that don't use it.

Maybe we'll need those types in a (far) future to implement some JDBC-style API, which exposes information about a database schema… But there is no reason to make life harder now because of this.

Reply:
As I understand from https://github.com/tarantool/tarantool-php/pull/164#discussion_r456708834 and similar, we want to save "unknown type" in field type for both (index part and field). Given this fact, is it necessary to have two arrays?

Reply:
I was sure that there are two arrays in tarantool, but it is not true. An index key part is validated against the same array of supported field types. The check that an index key part is not of 'any', 'array' or 'map' type is implemented separately (it is for 'HASH' and 'TREE'; 'RTREE' and 'BITSET' have more strict rules):

https://github.com/tarantool/tarantool/blob/8c50e069e3183222682896867c7e1ef98a5b99a7/src/box/memtx_space.c#L711-L723
https://github.com/tarantool/tarantool/blob/8c50e069e3183222682896867c7e1ef98a5b99a7/src/box/vinyl.c#L669-L680

BTW, while I looking into this, I found that our documentation may be a bit misleading and filed https://github.com/tarantool/doc/issues/1424

Your intention was to keep the code organized in the way similar to tarantool's code. So it looks meaningful to keep the single array of field types. Please, disregard my original comment.